Helper/Session.php

```diff
@@ -0,0 +1,168 @@
+        State $appState
+    ) {
+        parent::__construct($context);
+        $this->checkoutSession = $checkoutSession;
+        $this->adminCheckoutSession = $adminCheckoutSession;
+        $this->customerSession = $customerSession;
+        $this->logHelper = $logHelper;
+        $this->cache = $cache;
+        $this->appState = $appState;
+    }
+
+    /**
+     * Cache the session id for the quote
+     *
+     * @param int|string $parentQuoteId
+     * @param mixed $checkoutSession
+     */
+    public function saveSession($parentQuoteId, $checkoutSession) {
+        // cache the session id by parent quote id
+        $cacheIdentifier = self::BOLT_SESSION_PREFIX . $parentQuoteId;
+        $sessionData = [
+            "sessionType" => $checkoutSession instanceof \Magento\Checkout\Model\Session ? "frontend" : "admin",
+            "sessionID"   => $checkoutSession->getSessionId()
+         ];
+        $this->cache->save(serialize($sessionData), $cacheIdentifier, [], 86400);
+    }
+
+    /**
+     * Emulate session from cached session id
+     *
+     * @param mixed $session
+     * @param string $sessionID
+     */
+    private function setSession($session, $sessionID) {
+        // close current session
+        $session->writeClose();
+        // set session id (to value loaded from cache)
+        $session->setSessionId($sessionID);
+        // re-start the session
+        $session->start();
+    }
+
+    /**
+     * @param Quote $quote
+     * @return void
+     */
+    private function replaceQuote($quote) {
+        $this->checkoutSession->replaceQuote($quote);
+    }
+
+    /**
+     * Load logged in customer checkout and customer sessions from cached session id.
+     * Replace parent quote with immutable quote in checkout session.
+     *
+     * @param Quote $immutableQuote
+     * @throws \Magento\Framework\Exception\SessionException
+     */
+    public function loadSession($immutableQuote) {
+
+        // not an API call, no need to emulate session
+        if ($this->appState->getAreaCode() != \Magento\Framework\App\Area::AREA_WEBAPI_REST) {
+            $this->replaceQuote($immutableQuote);
+            return;
+        }
+
+        $customerId = $immutableQuote->getCustomerId();
+        $cacheIdentifier = self::BOLT_SESSION_PREFIX . $immutableQuote->getBoltParentQuoteId();
+
+        if ($serialized = $this->cache->load($cacheIdentifier)) {
+
+            $sessionData = unserialize($serialized);
+            $sessionID = $sessionData["sessionID"];
+
+            if ($sessionData["sessionType"] == "frontend") {
+                // shipping and tax, orphaned transaction
+                // cart belongs to logged in customer?
+                if ($customerId) {
+                    $this->setSession($this->checkoutSession, $sessionID);
+                    $this->setSession($this->customerSession, $sessionID);
+                }
+            } else {
+                // orphaned transaction
```

Comment:
orphan? admin?

Reply:
@daisy1754 why not? connection could be lost in the middle of communication the same as on frontend. tested. works.